src/libraries/Microsoft.Extensions.Configuration.Binder/gen/Model/KnownTypeSymbols.cs

```diff
@@ -108,6 +110,10 @@ public KnownTypeSymbols(CSharpCompilation compilation)
             IReadOnlyList_Unbound = compilation.GetBestTypeByMetadataName(typeof(IReadOnlyList<>))?.ConstructUnboundGenericType();
             IReadOnlySet_Unbound = compilation.GetBestTypeByMetadataName("System.Collections.Generic.IReadOnlySet`1")?.ConstructUnboundGenericType();
             ISet_Unbound = ISet?.ConstructUnboundGenericType();
+
+            // needed to be able to know if a member exist inside the compilation unit
+            Enum = compilation.GetBestTypeByMetadataName(typeof(Enum));
```

Comment:
Use the overload taking a `SpecialType` (i.e. `SpecialType.System_Enum`). All the [special types](https://learn.microsoft.com/en-us/dotnet/api/microsoft.codeanalysis.specialtype?view=roslyn-dotnet-4.6.0) are guaranteed to be in the input compilation.